src/include/kvs/kvs_handlers.hpp

```diff
@@ -92,4 +92,9 @@ std::pair<ReadCommittedPairLattice<std::string>, unsigned> process_get(
 void process_put(const Key& key, const unsigned long long& timestamp,
                  const std::string& value, Serializer* serializer,
                  std::unordered_map<std::string, unsigned>& key_size_map);
+
+bool is_primary_replica(const Key& key, std::unordered_map<Key, KeyInfo>& placement,
```

Comment:
I don't think we need to declare this globally. It's only used in `server.cpp`, so we can just define it locally there, no?

Reply:
but some other handlers (like `node_depart_handler`) in this header file are also used only by `server.cpp`?

Reply:
Right, but this is a helper function. Maybe it makes more sense to put it in `server_utils.hpp`?

Reply:
Probably, yeah.